validation/citrus-validation-json/src/test/java/org/citrusframework/validation/json/JsonElementValidatorTest.java

```diff
@@ -0,0 +1,272 @@
+                        "expected 'JSONArray'",
+                        "but was 'JSONObject'"
+                ),
+                new JsonAssertion(
+                        "",
+                        "{\"text\":\"Hello World!\", \"index\":5, \"id\":\"x123456789x\"}",
+                        "expected message contents, but received empty message"
+                )
+        ).toArray(new JsonAssertion[0]);
+    }
+
+    @Test(dataProvider = "validOnlyWithIgnoreExpressions")
+    public void shouldBeValidOnlyWithIgnoreExpressions(JsonAssertion jsonAssertion) {
+        var validationItem = toValidationItem(jsonAssertion);
+        fixture = new JsonElementValidator(NOT_STRICT, context, jsonAssertion.ignoreExpressions);
+        assertThatNoException().isThrownBy(() -> fixture.validate(validationItem));
+    }
+
+    @Test(dataProvider = "validOnlyWithIgnoreExpressions")
+    public void shouldBeInvalidWithoutIgnoreExpressions(JsonAssertion jsonAssertion) {
+        var validationItem = toValidationItem(jsonAssertion);
+        fixture = new JsonElementValidator(NOT_STRICT, context, Set.of());
+        assertThatThrownBy(() -> fixture.validate(validationItem)).isInstanceOf(ValidationException.class);
+    }
+
+    @DataProvider
+    public static JsonAssertion[] validOnlyWithIgnoreExpressions() {
+        return List.of(
+                new JsonAssertion(
+                        "{\"text\":\"Hello World!\", \"index\":5, \"object\":{\"id\":\"x123456789x\"}, \"greetings\":[{\"text\":\"Hello World!\", \"index\":1}, {\"text\":\"Hallo Welt!\", \"index\":2}, {\"text\":\"Hola del mundo!\", \"index\":3}],}",
+                        "{\"text\":\"Hello World!\", \"index\":\"?\", \"object\":{\"id\":\"?\"}, \"greetings\":\"?\"}",
+                        Set.of("$..index", "$.object.id", "$.greetings")
+                ),
+                new JsonAssertion(
+                        "{\"index\":\"bliblablu\"}",
+                        "{\"index\":\"tataa\"}",
+                        Set.of("$..index")
+                ),
+                new JsonAssertion(
+                        "{\"index\": {\"anything\": [0]} }",
+                        "{\"index\": {\"anything\": [55, 66, 77]} }",
+                        Set.of("$.index['anything'][*]")
+                )
+        ).toArray(new JsonAssertion[0]);
+    }
+
+    @Test
+    public void shouldBeValidWithIgnoreExpression() {
+        var jsonAssertion = new JsonAssertion(
+                "{\"index\": {\"something\": [0]} }",
+                "{\"index\": {\"anything\": [55, 66, 77]} }",
+                Set.of("$.index['anything'][*]")
+        );
+        var validationItem = toValidationItem(jsonAssertion);
+
+        fixture = new JsonElementValidator(NOT_STRICT, context, jsonAssertion.ignoreExpressions);
+
+        // TODO that shouldn't fail, should it?
```

Comment:
On second thought... 

`$.index['anything'][*]` should ignore all the children of `anything`, but the entry itself should be there nonetheless. Will update the test accordingly.

Reply:
I agree.